app/services/data_exports/process_part_service.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module DataExports
+  class ProcessPartService < BaseService
+    def initialize(data_export_part:)
+      super(nil)
+
+      @data_export_part = data_export_part
+      @data_export = data_export_part.data_export
+    end
+
+    def call
+      result.data_export_part = data_export_part
+
+      data_export_part.transaction do
+        # produce CSV lines into StringIO
+        export_result = data_export.export_class.call(data_export_part:).raise_if_error!
+        data_export_part.update!(csv_lines: export_result.csv_lines, completed: true)
+      end
+
+      # check if we are the last one to finish
+      if last_completed
+        DataExports::CombinePartsJob.perform_later(data_export_part.data_export)
+      end
```

Comment:
Why is the transaction needed? this only does one write in db.
Shall we use after_commit for the combine job queueing?

```suggestion
      # produce CSV lines into StringIO
      export_result = data_export.export_class.call(data_export_part:).raise_if_error!
      data_export_part.update!(csv_lines: export_result.csv_lines, completed: true)

      # check if we are the last one to finish
      if last_completed
        after_commit { DataExports::CombinePartsJob.perform_later(data_export_part.data_export) }
      end
```

Reply:
a remnant from previous iterations of this method, I've changed it to your suggestion 👍